clang/include/clang/Basic/Attr.td

```diff
@@ -4593,6 +4593,44 @@ def HLSLResourceBinding: InheritableAttr {
   let LangOpts = [HLSL];
   let Args = [StringArgument<"Slot">, StringArgument<"Space", 1>];
   let Documentation = [HLSLResourceBindingDocs];
+  let AdditionalMembers = [{
+      enum class RegisterType : unsigned { SRV, UAV, CBuffer, Sampler, C, I, Invalid };
```

Comment:
Do we need to have `Invalid` in this enum? I don't particularly like the implication that we still need to check the validity of the binding once we're dealing with the attribute we've applied for codegen.

Reply:
This enum is just moved from SemaHLSL but yeah, there is no reason to keep the Invalid value here. It is only used when validating the register letter.